test/e2e/projecttest/project_test.go

```diff
@@ -0,0 +1,327 @@
+						t.Error("expected output should include " + projName + projectId + ", in the out string ")
+					}
+				}
+			},
+		},
+
+		{
+			name: "project-delete-test",
+			fields: fields{
+				MainCmd:  projects.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command, rootCmd *cobra.Command, projectID string) {
+				root := c.Root()
+				RandName, _ := helper.GenerateRandomString(32)
+				projName := "metal-cli-" + RandName + "-project-delete-test"
+
+				projectId, _ := setupTestProject(t, projName)
+
+				if projectId != "" {
+					root.SetArgs([]string{subCommand, "delete", "-i", projectId, "-f"})
+					rescueStdout := os.Stdout
+					r, w, _ := os.Pipe()
+					os.Stdout = w
+					if err := root.Execute(); err != nil {
+						t.Error(err)
+					}
+					w.Close()
+					out, _ := io.ReadAll(r)
+					os.Stdout = rescueStdout
+					expectedOut := fmt.Sprintf("Project %s successfully deleted.", projectId)
+					if !strings.Contains(string(out[:]), expectedOut) {
+						t.Error(fmt.Errorf("expected output: '%s' but got '%s'", expectedOut, string(out)))
+					}
+				}
+			},
+		},
+
+		{
+			name: "project-bgpenbale-test",
+			fields: fields{
+				MainCmd:  projects.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command, rootCmd *cobra.Command, projectID string) {
+				root := c.Root()
+				RandName, _ := helper.GenerateRandomString(32)
+				projName := "metal-cli-" + RandName + "-project-bgpenable-test"
+
+				projectId, cleanProject := setupTestProject(t, projName)
+				defer cleanProject()
+				asn := "65000"
+				dtype := "local"
+
+				if projectId != "" {
+					root.SetArgs([]string{subCommand, "bgp-enable", "--project-id", projectId, "--deployment-type", dtype, "--asn", asn})
+					rescueStdout := os.Stdout
+					r, w, _ := os.Pipe()
+					os.Stdout = w
+					if err := root.Execute(); err != nil {
+						t.Error(err)
+					}
+					w.Close()
+					out, _ := io.ReadAll(r)
+					os.Stdout = rescueStdout
+					if !strings.Contains(string(out[:]), projectId) &&
+						!strings.Contains(string(out[:]), asn) &&
+						!strings.Contains(string(out[:]), dtype) {
+						t.Error("expected output should include " + projectId + "," + asn + " and " + dtype + ", in the out string ")
+					}
+				}
+			},
+		},
+
+		{
+			name: "project-bgpconfig-test",
+			fields: fields{
+				MainCmd:  projects.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command, rootCmd *cobra.Command, projectID string) {
+				root := c.Root()
+				RandName, _ := helper.GenerateRandomString(32)
```

Comment:
Rather than generating a different random string per sub-test, we should generated this once at the start of the `TestCli_`  function so that we use the same value across all sub-tests

Reply:
Also the variable name should start with a lowercase letter